Kernel/Output/HTML/Layout.pm

```diff
@@ -2919,6 +2920,50 @@ sub JSONReply {
     );
 }
 
+=head2 CustomException()
+
+=for stopwords customizable
+
+Throws an exception with customizable status code and body. Status can either be a client error (400-499) or a server error (500-599), see L<HTTP::Status> for details.
+
+    $LayoutObject->CustomException(
+        StatusCode => 500,           # optional, default: 500
+        Body       => 'Some text',   # optional, data to pass as response body
+    );
+
+=cut
+
+sub CustomException {
+    my ( $Self, %Param ) = @_;
+
+    # if given, check validity of status code
+    # allowed are client errors (400-499) and server errors (500-599)
+    my $StatusCode;
+    if ( $Param{StatusCode} ) {
+        if ( !is_client_error( $Param{StatusCode} ) && !is_server_error( $Param{StatusCode} ) ) {
+            $Kernel::OM->Get('Kernel::System::Log')->Log(
+                Priority => 'error',
+                Message  => "Status code $Param{StatusCode} is not a valid HTTP status!",
```

Comment:
The message could be more specific "is not an error HTTP status"

Reply:
Thank you :) Changed this.